satellite/satellitedb/overlaycache.go

```diff
@@ -22,6 +25,159 @@ type overlaycache struct {
+func (cache *overlaycache) FilterNodes(ctx context.Context, req *overlay.FilterNodesRequest) ([]*pb.Node, error) {
+	filterReputableReq := &filterReputableRequest{
+		minReputation:         req.MinReputation,
+		restrictions:          req.Restrictions,
+		excluded:              req.Excluded,
+		reputableNodeAmount:   req.ReputableNodeAmount,
+		newNodeAuditThreshold: req.NewNodeAuditThreshold,
+		// todo: make base configurable
+		base: 10,
+	}
+
+	reputableNodes, err := cache.filterReputableNodes(ctx, filterReputableReq)
+	if err != nil {
+		return nil, err
+	}
+
+	newNodeAmount := req.ReputableNodeAmount * int64(float64(req.ReputableNodeAmount)*req.NewNodePercentage)
+
+	filterNewReq := &filterNewRequest{
+		restrictions:          req.Restrictions,
+		excluded:              req.Excluded,
+		newNodeAmount:         newNodeAmount,
+		newNodeAuditThreshold: req.NewNodeAuditThreshold,
+		// todo: make base configurable
+		base: 10,
+	}
+
+	newNodes, err := cache.filterNewNodes(ctx, filterNewReq)
+	if err != nil {
+		return nil, err
+	}
+
+	var allNodes []*pb.Node
+	allNodes = append(allNodes, reputableNodes...)
+	allNodes = append(allNodes, newNodes...)
+
+	return allNodes, nil
+}
+
+func (cache *overlaycache) filterReputableNodes(ctx context.Context, req *filterReputableRequest) (reputableNodes []*pb.Node, err error) {
+	rows, err := cache.findReputableNodesQuery(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = utils.CombineErrors(err, rows.Close())
+	}()
+
+	for rows.Next() {
+		node := &dbx.Node{}
+		err = rows.Scan(&node.Id)
+		if err != nil {
+			return nil, err
+		}
+		id, err := storj.NodeIDFromBytes(node.Id)
+		if err != nil {
+			return nil, err
+		}
+		info, err := cache.Get(ctx, id)
+		if err != nil {
+			continue
+		}
+		reputableNodes = append(reputableNodes, info)
+	}
+
+	return reputableNodes, nil
+}
+
+func (cache *overlaycache) filterNewNodes(ctx context.Context, req *filterNewRequest) (newNodes []*pb.Node, err error) {
+	rows, err := cache.findNewNodesQuery(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = utils.CombineErrors(err, rows.Close())
+	}()
+
+	for rows.Next() {
+		node := &dbx.Node{}
+		err = rows.Scan(&node.Id)
+		if err != nil {
+			return nil, err
+		}
+		id, err := storj.NodeIDFromBytes(node.Id)
+		if err != nil {
+			return nil, err
+		}
+		info, err := cache.Get(ctx, id)
+		if err != nil {
+			continue
+		}
+		newNodes = append(newNodes, info)
+	}
+
+	return newNodes, nil
+}
+
+func (cache *overlaycache) findReputableNodesQuery(ctx context.Context, req *filterReputableRequest) (*sql.Rows, error) {
+	var auditCount int64
+	if req.minReputation.AuditCount > req.newNodeAuditThreshold {
+		auditCount = req.minReputation.AuditCount
+	} else {
+		auditCount = req.newNodeAuditThreshold
+	}
+
+	// todo: add filtering for free bandwidth and free disk space and not in excluded
+	rows, err := cache.db.Query(cache.db.Rebind(`SELECT nodes.id,
```

Comment:
replace all the arguments here with "?" and create an `args []interface` variable that you append all the arguments to. Then, call 
```
cache.db.Query(cache.db.Rebind(`SELECT .... `), args...)
```
Example at https://github.com/storj/storj/blob/master/satellite/satellitedb/statdb.go#L139-L143

Do the same with the other query.